tests/langs/mod.rs

```diff
@@ -73,763 +69,764 @@ impl InScopeLinePart {
 )]
 #[case(
     "base.c_variable",
     include_str!("c/base.c"),
-    C::new(CodeQuery::Prepared(PreparedCQuery::Variable)),
+    c::CompiledQuery::try_from(RawQuery::from(c::PreparedQuery::Variable)),
 )]
 #[case(
     "base.c_function",
     include_str!("c/base.c"),
-    C::new(CodeQuery::Prepared(PreparedCQuery::Function)),
+    c::CompiledQuery::try_from(RawQuery::from(c::PreparedQuery::Function)),
 )]
 #[case(
     "base.c_function_definition",
     include_str!("c/base.c"),
-    C::new(CodeQuery::Prepared(PreparedCQuery::FunctionDef)),
+    c::CompiledQuery::try_from(RawQuery::from(c::PreparedQuery::FunctionDef)),
 )]
 #[case(
     "base.c_function_declaration",
     include_str!("c/base.c"),
-    C::new(CodeQuery::Prepared(PreparedCQuery::FunctionDecl)),
+    c::CompiledQuery::try_from(RawQuery::from(c::PreparedQuery::FunctionDecl)),
 )]
 #[case(
     "base.c_switch",
     include_str!("c/base.c"),
-    C::new(CodeQuery::Prepared(PreparedCQuery::Switch)),
+    c::CompiledQuery::try_from(RawQuery::from(c::PreparedQuery::Switch)),
 )]
 #[case(
     "base.c_if",
     include_str!("c/base.c"),
-    C::new(CodeQuery::Prepared(PreparedCQuery::If)),
+    c::CompiledQuery::try_from(RawQuery::from(c::PreparedQuery::If)),
 )]
 #[case(
     "base.c_for",
     include_str!("c/base.c"),
-    C::new(CodeQuery::Prepared(PreparedCQuery::For)),
+    c::CompiledQuery::try_from(RawQuery::from(c::PreparedQuery::For)),
 )]
 #[case(
     "base.c_while",
     include_str!("c/base.c"),
-    C::new(CodeQuery::Prepared(PreparedCQuery::While)),
+    c::CompiledQuery::try_from(RawQuery::from(c::PreparedQuery::While)),
 )]
 #[case(
     "base.c_do",
     include_str!("c/base.c"),
-    C::new(CodeQuery::Prepared(PreparedCQuery::Do)),
+    c::CompiledQuery::try_from(RawQuery::from(c::PreparedQuery::Do)),
 )]
 #[case(
     "base.c_union",
     include_str!("c/base.c"),
-    C::new(CodeQuery::Prepared(PreparedCQuery::Union)),
+    c::CompiledQuery::try_from(RawQuery::from(c::PreparedQuery::Union)),
 )]
 #[case(
     "base.c_identifier",
     include_str!("c/base.c"),
-    C::new(CodeQuery::Prepared(PreparedCQuery::Identifier)),
+    c::CompiledQuery::try_from(RawQuery::from(c::PreparedQuery::Identifier)),
 )]
 #[case(
     "base.c_declaration",
     include_str!("c/base.c"),
-    C::new(CodeQuery::Prepared(PreparedCQuery::Declaration)),
+    c::CompiledQuery::try_from(RawQuery::from(c::PreparedQuery::Declaration)),
 )]
 #[case(
     "base.c_callexpr",
     include_str!("c/base.c"),
-    C::new(CodeQuery::Prepared(PreparedCQuery::CallExpression)),
+    c::CompiledQuery::try_from(RawQuery::from(c::PreparedQuery::CallExpression)),
 )]
 fn test_language_scopers(
     #[case] snapshot_name: &str,
     #[case] contents: &str,
-    #[case] lang: impl LanguageScoper,
+    #[case] lang: Result<impl LanguageScoper, TSQueryError>,
 ) {
+    let lang = lang.expect("Building a Language for a test should not fail");
```

Comment:
So there's no longer a way to statically (type level) say: custom, user-provided queries are expected to fail and if so, fail gracefully, while prepared queries are expected to be valid and would panic (but don't, as is validated through testing)?

Because under that model, `try_from` on a prepared query wouldn't exist.

Just asking. Not a blocker.

Reply:
Good point, we can definitely do more with typing given that we know the prepared queries are syntactically valid and tested to be so.

I'll give this a look and if you like the solution I'll squash the commits.

Reply:
Alright, but no worries. Such a refactor might lead down the path we came from in the first place (`PhantomData`, ...) 😉

And I'm not sure it was previously possible either, as the code query was an enum and not separate types.

Reply:
See what you think of something like this: 766ca786b45e50a8e67851dd44270953a8dbcd31.
Maybe a trait is strong enough.

We could seal the trait so that prepared queries are only definable internally.

Reply:
And I clearly need to fix the macro if we continue on with this solution.

Reply:
A new trait can work, but I'm wondering if the solution couldn't be quite simple:

```rust
impl From<PreparedQuery> for CompiledQuery {
    // ...
}
```

would that not be enough? This skips the intermediate, fallible `RawQuery` state, and skips to the known-good state (and panics if necessary, but tests will show it won't). It can live in parallel with `impl From<PreparedQuery> for RawQuery` if that's also needed.

Just skimmed over the proposal quickly so might be off base.

Reply:
And if it cannot be made simpler, I slightly prefer the previous approach, where we simply `unwrap` on `RawQuery`. A slight blip in the type system but a simpler implementation.